nixos/modules/services/misc/docker-registry.nix

```diff
@@ -114,6 +132,22 @@ in {
       };
     };
 
+    systemd.services.docker-registry-garbage-collect = {
+      description = "Run Garbage Collection for docker registry";
+
+      restartIfChange = false;
+      unitConfig.X-StopOnRemoval = false;
+
+      serviceConfig.Type = "oneshot";
+
+      script = ''
+        ${pkgs.docker-distribution}/bin/registry garbage-collect ${configFile}
+        ${lib.optionalString enableRedisCache '${pkgs.systemd}/bin/systemctl restart docker-registry'}
```

Comment:
single quotes won't work as single quotes can be part of variable names.
Use `''` for multiline strings or `"` for single line strings and you're safe %)